company-template.el

```diff
@@ -55,19 +56,44 @@
 
 (defun company-template-forward-field ()
   (interactive)
+  (let ((start (point))
+        (next-field-start (company-template-get-next-field-start)))
+    (push-mark)
+    (goto-char next-field-start)
+    (company-template-remove-field (company-template-field-at start))))
+
+(defun company-template-clear-field ()
+  "Clear field at point.
+If there is a next field, delete everything up to next field.  If field at
+point is last field, remove preceeding comma if present."
+  (interactive)
   (let* ((start (point))
-         (templates (company-template-templates-at (point)))
+         (current-ovl (company-template-field-at start)))
+    (when current-ovl
+      (let ((this-field-end (overlay-end current-ovl))
+            (next-field-start (company-template-get-next-field-start)))
+        (cond ((and (company-template-field-at next-field-start)
+                    (< this-field-end next-field-start)
+                    (string-match "^\\s-*,+\\s-*$" (buffer-substring-no-properties
```

Comment:
This should work fine, but I wonder if the regexps are more complex than the have to. Like, the template probably won't ever contain tabs, and there probably won't be any spaces before the comma.

Feel free to disregard this comment.